USM.Test/AlgorithmTest.cs

```diff
@@ -0,0 +1,509 @@
+            //adding for cse 22
+            CSE22.Sections.Add(CSE22Section1);
+            CSE22Section1.Meetings.Add(CSE22Lecture);
+            //adding for cse 23
+            CSE23.Sections.Add(CSE23Section1);
+            CSE23Section1.Meetings.Add(CSE23Lecture);
+            //adding for cse 24
+            CSE24.Sections.Add(CSE24Section1);
+            CSE24Section1.Meetings.Add(CSE24Lecture);
+            //adding for cse 25
+            CSE25.Sections.Add(CSE25Section1);
+            CSE25Section1.Meetings.Add(CSE25Lecture);
+            //adding for cse 26
+            CSE26.Sections.Add(CSE26Section1);
+            CSE26Section1.Meetings.Add(CSE26Lecture);
+            CSE26Section1.Meetings.Add(CSE26Discussion1);
+            CSE26Section2.Meetings.Add(CSE26Lecture);
+            CSE26Section2.Meetings.Add(CSE26Discussion2);
+            CSE26.Sections.Add(CSE26Section2);
+            //adding for cse 27
+            CSE27.Sections.Add(CSE27Section1);
+            CSE27Section1.Meetings.Add(CSE27Lecture);
+            CSE27Section1.Meetings.Add(CSE27Discussion1);
+            CSE27Section2.Meetings.Add(CSE27Lecture);
+            CSE27Section2.Meetings.Add(CSE27Discussion2);
+            CSE27.Sections.Add(CSE27Section2);
+
+            Course[] courses = { CSE20, CSE21, CSE22, CSE23, CSE24, CSE25, CSE26, CSE27 };
+            return courses;
+        }
+
+        //desperate single test
+        [Fact]
+        public void TestSingleClass()
+        {
+            //use the course above
+            ScheduleRepo sch = new ScheduleRepo();
+            Course[] AllCourses = Classes();
+
+            // choose the courses to be tested
+            Course CSE20 = AllCourses[0];
+            //Course CSE21 = AllCourses[1];
+            Course[] courses = { CSE20 };
+
+            //call the algorithm
+            List<List<Section>> returned = sch.FindScheduleForClasses(courses);
+            int expectedNum = 1;
+            int actualNum = returned.Count;
+
+            Assert.Equal(expectedNum, actualNum);
+        }
+
+        /* conflict on the same day and same time */
+        [Fact]
+        public void TestConflictSameDay()
+        {
+            //use the course above
+            ScheduleRepo sch = new ScheduleRepo();
+            Course[] AllCourses = Classes();
+
+            // choose the courses to be tested
+            Course CSE20 = AllCourses[0];
+            Course CSE21 = AllCourses[1];
+            Course[] courses = { CSE20, CSE21 };
+
+            //call the algorithm
+            List<List<Section>> returned = sch.FindScheduleForClasses(courses);
+            int expectedNum = 0;
+            int actualNum = returned.Count;
+
+            //no available schedules
+            Assert.Equal(expectedNum, actualNum);
+        }
+
+        //conflict on same day different time
+        [Fact]
+        public void TestConflictSameDayDifTime()
+        {
+            //use the course above
```

Comment:
This assumes that this method follows a specific method, so if the orders were changed the test wouldn't run as expected.  Make the courses/test cases more independent.